MahApps.Metro/Controls/Dialogs/CustomInputDialog.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Threading.Tasks;
+using System.Windows;
+
+namespace MahApps.Metro.Controls.Dialogs
+{
+    public abstract class CustomInputDialog<T> : BaseMetroDialog where T : class {
+        public abstract Task<T> WaitForButtonPressAsync(); 
+
+        protected override void OnLoaded()
+        {
+            this.AffirmativeButtonText = this.DialogSettings.AffirmativeButtonText;
+            this.NegativeButtonText = this.DialogSettings.NegativeButtonText;
+        }
+
+        public static readonly DependencyProperty MessageProperty = DependencyProperty.Register("Message", typeof(string), typeof(CustomInputDialog<T>), new PropertyMetadata(default(string)));
+        public static readonly DependencyProperty InputProperty = DependencyProperty.Register("Input", typeof(TaskCompletionSource<T>), typeof(CustomInputDialog<T>), new PropertyMetadata(default(T)));
```

Comment:
@jords1987 there is no property for `InputProperty`? and is `typeof(TaskCompletionSource<T>)` the correct value for the type?